engine/src/main/java/org/hibernate/validator/internal/metadata/core/MetaConstraint.java

```diff
@@ -41,13 +45,28 @@
 	private final ConstraintLocation location;
 
 	/**
+	 * The {@link ValueExtractor} used to extract the value for validation.
+	 */
+	private final ValueExtractorDescriptor valueExtractorDescriptor;
```

Comment:
Ah, ok, so that's already planning for allowing different validation-applies-to settings for the constraint of one element. Makes sense.